### ISSUE TITLE
[HOLD][$250] Dev - Getting console warning in android

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#11733]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. build the app
2. immediately see the errors

## Expected Result:
No console warnings displayed

## Actual Result:
 Console warnings about `keyboardAvoidingViewBehavior` are present

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Android

**Version Number:** 1.2.12-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
![2022-10-11_11-14-40](https://user-images.githubusercontent.com/43996225/195209379-34bbaca4-0085-40a9-886d-39032268a873.png)
![2022-10-11_11-14-48](https://user-images.githubusercontent.com/43996225/195209382-c4ab3fca-30c4-4d76-a281-547313c76ba0.png)

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @aneequeahmad 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1665522948008369

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by hungvu193** [comment]
Actually, change the default proptypes can effects other screen.
For this case I think we should change:
https://github.com/Expensify/App/blob/299f07bfa29c0dd81a6126691aae29e161097baa/src/pages/home/ReportScreen.js#L210
into:
```
 keyboardAvoidingViewBehavior={Platform.OS === 'android' ? undefined : 'padding'} 
```

**Comment by marcaaron** [comment]
Please wait until the issue has been triaged and `Help Wanted` label added.

**Comment by marcaaron** [comment]
>I vote we close this issue as it's not a big problem

I am not opposed to waiting if there's an open PR where it will be fixed, but let's leave this issue open until it is fixed. It's a problem, that's why I reported it, and we should fix it.

**Comment by parasharrajat** [comment]
Strangely, that component is using `Platform API`.  We are discussing to revert it in https://github.com/Expensify/App/pull/10648. I am not sure when that will be done. 

@marcaaron Do you want me to look into proposals or hold this?

**Comment by marcaaron** [comment]
If we can fix this now it is fine to just do it.

**Comment by marcaaron** [comment]
We should generally not be merging PRs that introduce these kinds of warnings.

**Comment by parasharrajat** [comment]
I agree. Looks like there was discussion on the same https://github.com/Expensify/App/pull/10567/files#r955053952.

**Comment by parasharrajat** [comment]
@aimane-chnaif 's [proposal](https://github.com/Expensify/App/issues/11733#issuecomment-1275544293) looks good to me. 

That line looks unnecessary to me as well. 

cc: @marcaaron 

:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by marcaaron** [comment]
cc @Luke9389 as I am not the assigned CME

**Comment by parasharrajat** [comment]
Oh Sorry, Didn't notice.

**Comment by Luke9389** [comment]
I think @aimane-chnaif's proposal to set the android value to `height` will cause [this issue](https://github.com/Expensify/App/issues/10539) again (that's why we made[ this PR](https://github.com/Expensify/App/pull/10567))

**Comment by parasharrajat** [comment]
So the android value is already set to `height`, it is just removing that warning.

**Comment by hungvu193** [comment]
> I think @aimane-chnaif's proposal to set the android value to `height` will cause [this issue](https://github.com/Expensify/App/issues/10539) again (that's why we made[ this PR](https://github.com/Expensify/App/pull/10567))

that's why I think we should leave it to `undefined`

**Comment by hungvu193** [comment]
@aimane-chnaif No, I don't think so, let take a look at KeyboardAvoidingView component at react native repo:
https://github.com/facebook/react-native/blob/8cdc9e7f04e2dd3026d15dcd5765952bfb2c5c08/Libraries/Components/Keyboard/KeyboardAvoidingView.js#L181-L243
```javascript
switch (behavior) {
      case 'height':
        let heightStyle;
        if (this._frame != null && this.state.bottom > 0) {
          // Note that we only apply a height change when there is keyboard present,
          // i.e. this.state.bottom is greater than 0. If we remove that condition,
          // this.frame.height will never go back to its original value.
          // When height changes, we need to disable flex.
          heightStyle = {
            height: this._initialFrameHeight - bottomHeight,
            flex: 0,
          };
        }
        return (
          <View
            ref={this.viewRef}
            style={StyleSheet.compose(style, heightStyle)}
            onLayout={this._onLayout}
            {...props}>
            {children}
          </View>
        );

      case 'position':
        return (
          <View
            ref={this.viewRef}
            style={style}
            onLayout={this._onLayout}
            {...props}>
            <View
              style={StyleSheet.compose(contentContainerStyle, {
                bottom: bottomHeight,
              })}>
              {children}
            </View>
          </View>
        );

      case 'padding':
        return (
          <View
            ref={this.viewRef}
            style={StyleSheet.compose(style, {paddingBottom: bottomHeight})}
            onLayout={this._onLayout}
            {...props}>
            {children}
          </View>
        );

      default:
        return (
          <View
            ref={this.viewRef}
            onLayout={this._onLayout}
            style={style}
            {...props}>
            {children}
          </View>
        );
    }
  }
}
```

In this case undefined and "" will have the same effect, but if you use undefined it won't  throw the error because it is included in proptypes:
```
 behavior?: 'height' | 'position' | 'padding' | undefined;
```

**Comment by hungvu193** [comment]
@aimane-chnaif Ah I see. Since I don't want keyboardAvoidingViewBehavior affect the app, because it changed the value of behavior, I think the best way to do that is adding another props: `enable` and set default to `true`.
```javascript
keyboardAvoidingViewEnable: PropTypes.bool,
```
```javascript
const defaultProps = {
    style: [],
    includePaddingBottom: true,
    includePaddingTop: true,
    onTransitionEnd: () => {},
    modal: {},
    keyboardAvoidingViewBehavior: 'padding',
    keyboardAvoidingViewEnable: true
};
```
and disable it in ReportScreen:

```javascript
 <ScreenWrapper
                style={[styles.appContent, styles.flex1, {marginTop: this.state.viewportOffsetTop}]}
                keyboardAvoidingViewBehavior={Platform.OS === 'android' ? undefined : 'padding'}
                keyboardAvoidingViewEnable={false}
            >
```

**Comment by parasharrajat** [comment]
@Luke9389 Thanks. Now, I see the motive behind that PR. let me check this quickly.

**Comment by parasharrajat** [comment]
Ok, I checked again. It will indeed cause that issue. I retract my decision. I like the suggestion from @hungvu193 in https://github.com/Expensify/App/issues/11733#issuecomment-1277010019 to add `enable/disable` behavior but there is another PR that is trying to change this code. 

These are two options here. 
1. Either we refactor the code following our guidelines and add enable/disable functionality.https://github.com/Expensify/App/issues/11733#issuecomment-1277010019
2. Just fix the warning. https://github.com/Expensify/App/issues/11733#issuecomment-1275546645

In both ways, @hungvu193 proposals make sense. 


cc: @Luke9389 what do you think we should do?

IMO, let's just focus on point 2 as the issue says. There are some ongoing efforts in different PR/issues which will affect the KeyboardAvoidingView behavior.

:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by Luke9389** [comment]
@aimane-chnaif, please be careful here. [There is a known issue](https://github.com/Expensify/App/issues/10539) that we've linked to that is caused by using `height`. Are you really 100% positive that you checked that problem? 

I do agree that we should wait on [this PR](https://github.com/Expensify/App/pull/11586) to be merged. If the issue is still happening, I'd prefer to go with option 2 from @parasharrajat's comment [here](https://github.com/Expensify/App/issues/11733#issuecomment-1277902834). @parasharrajat I don't like option 1 very much because it seems like an anti-pattern to have a component enable or disable itself (as opposed to just not using it in the parent). 

For now I'd like to put this issue on HOLD to wait for that PR. Thoughts @parasharrajat?

**Comment by parasharrajat** [comment]
fine with me.

**Comment by Luke9389** [comment]
On HOLD.

**Comment by aneequeahmad** [comment]
I think i should also be added in `reporter` as I reported this bug on 1st September [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1662053447564539) and again same error on another component on 28th September [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1664319888468459?thread_ts=1662053447.564539&cid=C01GTK53T8Q). Thanks

cc: @kavimuru, @MitchExpensify, @Luke9389

**Comment by MitchExpensify** [comment]
Looks fair to me @aneequeahmad!

**Comment by MitchExpensify** [comment]
I'm going to hold on the Upwork job while the issue is on hold

**Comment by Luke9389** [comment]
still on hold

**Comment by MitchExpensify** [comment]
@Luke9389 What do you think about closing this given the above comment?

**Comment by Luke9389** [comment]
Gotta read this one back over.

**Comment by Luke9389** [comment]
Yep we can close this now. Thanks @aimane-chnaif.

**Comment by aneequeahmad** [comment]
@MitchExpensify  This one is fixed in production, am i eligible for reporting bonus ?

**Comment by MitchExpensify** [comment]
Yes @aneequeahmad, paying now

**Comment by MitchExpensify** [comment]
Mind [applying here](https://www.upwork.com/jobs/~0131f45457604e2be4) and I'll issue the reporting bonus? Thanks @aneequeahmad

**Comment by aneequeahmad** [comment]
@MitchExpensify applied thanks

**Comment by MitchExpensify** [comment]
Paid!
